data/abi/abi_encode_test.go

```diff
@@ -1264,4 +1264,20 @@ func TestInferToSlice(t *testing.T) {
 		require.NoError(t, err, "inferToSlice on testcase %d failed to successfully infer %v", i, test.toBeInferred)
 		require.Equal(t, test.length, len(inferredSlice), "inferToSlice on testcase %d inferred different length, expected %d", i, test.length)
 	}
+
+	// one more testcase for totally nil (with no type information) is bad, should not pass the test
+	_, err := inferToSlice(nil)
+	require.Error(t, err, "infer to Slice should not accept nil interface")
+	require.EqualError(
```

Comment:
We can skip the `require.Error` since it's implied by the `EqualError`, right?

Reply:
yes, that is correct